glotaran/project/scheme.py

```diff
@@ -50,6 +50,10 @@ class Scheme:
     saving: SavingOptions = SavingOptions()
     result_path: str | None = None
 
+    def __post_init__(self):
```

Comment:
If there's a single dataset and the group_tolerance is 0 then grouping should be False.

Reply:
This is a user overrideable and also a single dataset can be calculated as a grouped problem, this is crucial for tests